pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -881,8 +908,9 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 	return volume, err
 }
 
+// getComputeIDbyHostname tries to find ID of the OpenStack instance where
+// this code runs.
 func (os *OpenStack) getComputeIDbyHostname(cClient *gophercloud.ServiceClient) (string, error) {
```

Comment:
I know this isn't part of your change, but this entire function confuses me.  Why do we need to fetch *all* instances and then matching hostnames on the client side?

I would have expected this function to be something as simple as (with a few more lines of error handling):
```
   hostname, err := ossys.Hostname()
   i, ok := os.Instances()
   return i.InstanceID(hostname)
```